### ISSUE TITLE
How do I generate a dart client for custom nextcloud app?

### ISSUE THREAD

**Issue by samolego** [nextcloud/neon#2202]
Are there any guides on this? I'd like to generate a dart client from an existing nextcloud app - is that possible?

**Comment by samolego** [comment]
Ah, ok. This one: https://memories.gallery/

**Comment by samolego** [comment]
I'd like to try create a native flutter app, as the default memories one is a bit slow on mobile devices

**Comment by provokateurin** [comment]
We also have plans to implement a Photos app that can also work with Memories as a backend (https://github.com/nextcloud/neon/issues/355), so you might want to consider using our framework for building your app.
For a prototype it would be fine to use a hand-written specification, see the discussion in https://github.com/nextcloud/neon/issues/944 for reference.

**Comment by samolego** [comment]
Ok, thanks!

**Comment by samolego** [comment]
Sorry to bother again, I've generated a openapi.yaml from observing the traffic.
Do I have to use dynamite package now?

**Comment by provokateurin** [comment]
Yes! You should drop the file into packages/nextcloud/lib/src/api and name it memories.openapi.yaml. Then you can run the generate script using ./tool/generate-nextcloud.sh (from the root of the project). Afterwards you should have 3 new dart files.
What tool did you use to generated the specification? I wasn't aware something like that exists.

**Comment by samolego** [comment]
Well, I navigated around the app, used firefox's save traffic to har file, then used https://github.com/jonluca/har-to-openapi to convert har to openapi.yml.

Not sure if it's correct, first time doing it.
